pydantic_ai_slim/pydantic_ai/messages.py

```diff
@@ -112,12 +120,14 @@ def __init__(
         self,
         url: str,
         force_download: bool = False,
+        identifier: str | None = None,
         vendor_metadata: dict[str, Any] | None = None,
         media_type: str | None = None,
     ) -> None:
         self.url = url
-        self.vendor_metadata = vendor_metadata
         self.force_download = force_download
+        self.identifier = identifier
```

Comment:
Can you see if we can move the `multi_modal_content_identifier` function that's used in `_agent_graph.py` to here, so we can use it if no `identifier` was explicitly provided? That way we may be able to define the field as always being `str` (so no `None`) and always use it without having to check if it's set, similar to `media_type`.